src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,48 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MAX_AGE = 100;
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH_OF_PASSWORD = 6;
+    private static final String EMPTY = "";
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("User cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login can't be null");
+        }
+        if (user.getLogin().equals(EMPTY)) {
+            throw new InvalidDataException("Login can't be empty");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("This person already exist");
+        }
+        if (user.getAge() == null) {
+            throw new InvalidDataException("Age can't be null");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new InvalidDataException("You can't register children");
+        }
+        if (user.getAge() > MAX_AGE) {
+            throw new InvalidDataException("You are too old, just have a rest");
```

Comment:
I think you should to make much more understandable messages in Exception construction))
```suggestion
            throw new InvalidDataException("User age is bigger than max age");
```